### PR TITLE
Purge cluster fixes

### PR DESCRIPTION
**Pull request by alfredodeza**
These fixes are in master but were never ported back to the stable-2.1 branch.

Reference: https://bugzilla.redhat.com/show_bug.cgi?id=1413985

### CODE REVIEW

File: infrastructure-playbooks/purge-cluster.yml [hunk 1]
```diff
@@ -408,6 +400,14 @@
      ceph_data_partlabels.rc == 0 and
      zap_block_devs
 
+  - name: get ceph journal partitions
+    shell: |
+      blkid | awk '/ceph journal/ { sub (":", "", $1); print $1 }'
+    when:
+      - ceph_journal_partlabels.rc == 0
```
Comment (leseb) [inline infrastructure-playbooks/purge-cluster.yml]: With a single line, we usually do it like this: `when: ceph_journal_partlabels.rc == 0` only.
(yes I know, **I** did that...)

**Comment by andrewschoen** [comment]
@leseb That sounds like a reasonable solution, I'm not sure we can trust "Since we import group_vars and defaults" though, I've seen those includes not work and cause unexpected failures.

I think would should cherry-pick these commits unmodified from master and then if we want to clean this up it should be in a separate PR against the master branch. Thoughts?

**Comment by leseb** [comment]
@andrewschoen right let's just do cherry-picks here and in a separate PR we will address the refactor.

**Comment by andrewschoen** [comment]
I'm going to go ahead and merge this and we'll address @leseb's concern in another PR.

**Comment by alfredodeza** [comment]
@leseb agree on improving/changing things somewhere else, this is just a backport.

You do mention however that `zap_block_devs` should be used. I don't think a user should need to know about that flag when purging a cluster to be able to purge correctly. It is *nice* to have it around for granular control though.

The variable will probably need to be renamed as it implies it will be used to "zap" a "block device" but it is used as a conditional for the task: "zap ceph journal partitions"

**Comment by leseb** [comment]
@alfredodeza right, let's address this in another PR. Let me have a look at that.

**Pull request merged**